### DISCUSSION TITLE
New Plugin type: Value Formatter

### DISCUSSION THREAD

**Discussion by bobfrankly** [grafana/grafana#37978] [Plugins]
I see plugin options for Panels, and for frontend/backend data sources. I also see a few questions about formatting values. What I do not see is an option for extending the valueFormats.

Digging through the source, a value formatter appears to be:
1. a function
2. an ID
3. and a display friendly name

These are seemingly declared in the valueFormats folder alongside the [exporting definitions](https://github.com/grafana/grafana/blob/d7c76dacad0b5b867acc53d476bdb5664327e26b/packages/grafana-data/src/valueFormats/valueFormats.ts), which is then exposed to the user [via this bit](https://github.com/grafana/grafana/blob/d7c76dacad0b5b867acc53d476bdb5664327e26b/packages/grafana-ui/src/components/UnitPicker/UnitPicker.tsx).

I could be totally wrong on this and I welcome corrections, my js/ts skills are nothing to write home about. However my point is that these seems pretty simple to create, the userbase (_myself among them_ 😁 )  just need a way to add to them via the plugin channels. Then we can write the value formatters we need and use/share the plugins, which could also help clear out some other lingering issues from the Github. 😛 What are the chances of this happening?

**Comment by sympatheticmoose** [comment]
I'm curious if theres reasons why these formatters shouldn't just be added to Grafana itself? Has there been issues contributing them?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
